config/auth_metadata_service.go

```diff
@@ -0,0 +1,169 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/logger"
+	"golang.org/x/oauth2"
+)
+
+// timeout to wait for metadata service to respond
+const metadataServiceTimeout = 10 * time.Second
+
+const MetadataServiceVersion = "1"
+const MetadataServiceVersionHeader = "X-Databricks-Metadata-Version"
+const MetadataServiceHostHeader = "X-Databricks-Host"
+
+// response expected from the metadata service
+type serverResponse struct {
+	AccessToken string `json:"access_token"`
+	TokenType   string `json:"token_type"`
+
+	// epoch in seconds since 1970-01-01T00:00:00Z
+	ExpiresOn json.Number `json:"expires_on"`
+}
```

Comment:
lines 24-30 are duplicates of https://github.com/databricks/databricks-sdk-go/blob/main/config/auth_azure_msi.go#L153-L157